### PR TITLE
feat: display user saved lists

### PR DESCRIPTION
**Pull request by kuck1**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced components for managing and displaying saved lists of organizations and services.
	- Added localization support with new strings for improved user interactions and notifications.
	- Enhanced UI with responsive layouts and loading states for a better user experience.
	- Added a customizable confirmation modal for delete actions to improve user interaction.

- **Bug Fixes**
	- Improved data synchronization to ensure the UI reflects the latest state of saved items, including timestamps for last updates.

- **Style**
	- Introduced a new button variant for enhanced UI design flexibility and improved hover effects.

### CODE REVIEW

**Comment by JoeKarow** [comment]
@coderabbitai review

File: apps/app/public/locales/en/common.json [hunk 1]
```diff
@@ -171,7 +171,16 @@
 		"name": "List name",
 		"new-list-placeholder": "Enter list name...",
 		"removedMulti": "Resource removed from list: {{name}}",
-		"removedSingle": "Resource removed from list"
+		"removedSingle": "Resource removed from list",
+		"saved": "Saved",
```
Comment (kuck1) [inline apps/app/public/locales/en/common.json:175]: Use 'word.saved'

File: apps/app/src/pages/account/saved/[listId].tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+import { api } from '~app/utils/api'
+import { getServerSideTranslations } from '~app/utils/i18n'
+import { Icon } from '~ui/icon'
+
+// @ts-expect-error Next Dynamic doesn't like polymorphic components
+const QuickPromotionModal = dynamic(() =>
+	import('@weareinreach/ui/modals/QuickPromotion').then((mod) => mod.QuickPromotionModal)
+)
+
+const formatDate = (dateString: Date) => {
```
Comment (JoeKarow) [inline apps/app/src/pages/account/saved/[listId].tsx:24]: Use 'luxon' for date formatting

File: apps/app/src/pages/account/saved/[listId].tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+	const { data: session, status } = useSession()
+	const router = useRouter<'/account/saved/[listId]'>()
+	const { listId } = router.query
+	const handleReturnHome = useCallback(() => router.replace('/'), [router])
+	const {
+		data: queryResult,
+		isLoading,
+		refetch,
+	} = api.savedList.getById.useQuery({ id: listId ?? '' }, { enabled: !!listId })
+
```
Comment (kuck1) [inline apps/app/src/pages/account/saved/[listId].tsx:56]: `const apiUtils = api.useUtils()`

File: apps/app/src/pages/account/saved/index.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import { getServerSideTranslations } from '~app/utils/i18n'
+import { SavedResultLoading } from '~ui/components/core/Saved/SavedOrgResultCard'
+import { Icon } from '~ui/icon'
+import { CreateNewList } from '~ui/modals'
+
+const QuickPromotionModal = dynamic(() =>
+	import('@weareinreach/ui/modals/QuickPromotion').then((mod) => mod.QuickPromotionModal)
+)
+
+const formatDate = (dateString: Date) => {
```
Comment (kuck1) [inline apps/app/src/pages/account/saved/index.tsx:22]: use `luxon` for date formatting/display

File: apps/app/src/pages/account/saved/index.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	return (
+		<Grid.Col xs={12} sm={12}>
+			<Stack>
+				<Title order={1}> {t('list.saved')} </Title>
+				<Text size='lg'>{t('list.create-new-sub')}</Text>
+				<CreateNewList
+					style={{
+						fontSize: '16px',
```
Comment (kuck1) [inline apps/app/src/pages/account/saved/index.tsx:83]: move styles in to a useStyles() hook - see other components for how `createStyles()` is used from `@mantine/core`

File: packages/api/router/savedLists/index.ts [hunk 6]
```diff
@@ -13,7 +13,7 @@ export const savedListRouter = defineRouter({
 		return handler(opts)
 	}),
 	/** Get list by ID. List must be owned by or shared with logged in user */
-	getById: protectedProcedure.input(schema.ZGetByIdSchema).query(async (opts) => {
+	getById: protectedProcedure.input(schema.ZGetByIdInputSchema).query(async (opts) => {
```
Comment (kuck1) [inline packages/api/router/savedLists/index.ts:16]: revert back to original

File: packages/api/router/savedLists/query.getById.schema.ts [hunk 7]
```diff
@@ -2,5 +2,8 @@ import { z } from 'zod'
 
 import { prefixedId } from '~api/schemas/idPrefix'
 
-export const ZGetByIdSchema = z.object({ id: prefixedId('userSavedList') })
-export type TGetByIdSchema = z.infer<typeof ZGetByIdSchema>
+export const ZGetByIdInputSchema = z.object({
```
Comment (kuck1) [inline packages/api/router/savedLists/query.getById.schema.ts:5]: revert back to original

File: packages/ui/components/core/ActionButtons/Save.tsx [hunk 8]
```diff
@@ -69,17 +69,21 @@ const ListItem = ({ data, name, action }: ListMenuProps) => {
 		displayText: t('list.error-remove'),
 	})
 
+	const { data: allSavedLists, refetch } = api.savedList.getAll.useQuery()
+
 	const saveItem = api.savedList.saveItem.useMutation({
 		onSuccess: (_, { itemId }) => {
 			savedInList()
 			utils.savedList.isSaved.invalidate(itemId)
+			refetch()
```
Comment (kuck1) [inline packages/ui/components/core/ActionButtons/Save.tsx:78]: change to invalidate

**Comment by kuck1** [comment]
@coderabbitai review

**Comment by kuck1** [comment]
@coderabbitai review 1343

File: packages/ui/components/core/Saved/SavedOrgResultCard.tsx [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+		() =>
+			savedItem.communityBadges.length ? (
+				<Badge.Group>
+					{savedItem.communityBadges.map(({ icon, tsKey }) => (
+						<Badge.Community icon={icon ?? ''}>{t(tsKey, { ns: 'attribute' })}</Badge.Community>
+					))}
+				</Badge.Group>
+			) : null,
+		[savedItem, t]
+	)
```
[the remark this answers is not in the document]
Comment (kuck1) [reply packages/ui/components/core/Saved/SavedOrgResultCard.tsx]: @coderabbitai I just pushed the change for this

**Comment by kuck1** [comment]
@coderabbitai summary

**Comment by kuck1** [comment]
Hey @JoeKarow I think we're getting pretty close with this one

**Comment by kuck1** [comment]
@coderabbitai summary

**Review by JoeKarow — APPROVED** [review]

**Comment by JoeKarow** [comment]
@coderabbitai resolve

**Pull request merged**